src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerProcessor.cs

```diff
@@ -0,0 +1,200 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement.ObjectTypes.Server
+{
+     /// <summary>
+    /// Class to manage affinity for 64 processors in an independent manner for I/O as well as processors
+    /// </summary>
+    internal sealed class AffinityMngr
+    {
+        private const int MAX32CPU = 32;
+        private static string[] configFields = new string[]
+            {
+                "Minimum",
+                "Maximum",
+                "Dynamic",
+                "ConfigValue",
+                "RunValue"
+            };
+        #region class Affinity representing data related to each affinity mask
+        /// <summary>
+        /// The Affinity structure represents one row of the sys.configuration table for
+        /// that particular affinity (affinity,affinty64 for processor or I/O).
+        /// It also stores the old value of the affinity for optimizing server query.
+        /// </summary>
+        private class Affinity
+        {
+
+            private int affinityMaskCfg = 0;
+            private int affinityMaskRun = 0;
+            /// <summary>
+            /// Affinity ConfigValue reflecting sys.configurations table
+            /// </summary>
+            public int AffinityMaskCfg { get { return affinityMaskCfg; } set { affinityMaskCfg = value; } }
+            /// <summary>
+            /// Affinity RunValue reflecting sys.configurations table
+            /// </summary>
+            public int AffinityMaskRun { get { return affinityMaskRun; } set { affinityMaskRun = value; } }
+
+            /// Reset the affinities to default values.
+            /// </summary>
+            public void Clear()
+            {
+                affinityMaskCfg = 0;
+                affinityMaskRun = 0;
+            }
+        };
+        #endregion
+
+        /// <summary>
+        /// Generates a mask for masking the N processors
+        /// </summary>
+        /// <param name="numProcessors"> number of processors to mask</param>
+        /// <returns>Mask for numProcessors</returns>
+        private int GetMaskAllProcessors(int numProcessors)
+        {
+            System.Diagnostics.Debug.Assert(numProcessors > 0 && numProcessors <= 32, "Masking all processors failed!");
```

Comment:
Remove these Debug asserts - either convert them into actual thrown exceptions if it's something that isn't actually supported, or ignore/log as appropriate.

(same applies to all added in this PR)